### DISCUSSION TITLE
How to modify an asset key in runtime?

### DISCUSSION THREAD

**Discussion by 4l1fe** [dagster-io/dagster#20430] [Q&A]
Hey.

I would like to change an asset key to propogate its value to dumping to a filesystem. Particularly, `key_prefix` to keep all the `AssetKey`'s machinary working.

Please, direct me to a way to achieve it. Fow now, I'm trying to redefine the method `UPathIOManager._get_path(self, context)` having custom data in `context` and extending the final variable `path` with it.

**Comment by maximearmstrong** [comment]
@4l1fe Could you tell me more about your use case? Is it related to this [request](https://github.com/dagster-io/dagster/discussions/20413)?

I don't recommend modifying an asset key or its prefix anywhere else than in your definitions. Perhaps with more details on your use case, we could find another solution.

**Reply by 4l1fe** [reply]
Sry for the late reply. Yeah, it is related. The case is simple, make a job run explicitly related to a user id.

For now, i understand Dagster internals a bit better and decided not to modify filesystem hierarchy, but inject a user id into a result. I think i should follow the way Dagster makes data dumps to keep them maintainable.

The biggest insight i got is that Asset isn't that abstraction i have to use for my purpose, but Op is. 🙂
